File: programmers/해시/위장.py
def solution(clothes):
    answer = 1

    items = {}

    # 모든 items 딕셔너리 생성
    for thing in clothes:
        key = thing[1]
        value = thing[0]
        if key in items:
            items[key].append(value)
        else:
            items[key] = [value]

    # 가능한 모든 조합 구하기
    for key in items.keys():
        answer *= len(items[key]) + 1

    return answer - 1

File: programmers/해시/위장.java
import java.util.*;
import java.io.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;

        Map<String, Integer> map = new HashMap<>();

        for(String[] c:clothes){
            if(map.containsKey(c[1])){
                int value = map.get(c[1]);
                map.put(c[1],value+1);
            } else{
                map.put(c[1],1);
            }
        }
        // System.out.println(map);

        for(String key:map.keySet()){
            answer*=map.get(key)+1;
        }

        return answer-1;
    }
}
